modules/claims_api/app/models/claims_api/form_526.rb

```diff
@@ -68,10 +69,61 @@ def to_internal
       }.to_json
     end
 
+    def current_mailing_address
+      veteran[:currentMailingAddress]
+    end
+
     private
 
     def cleaned_errors
       errors.messages.map { |key, value| "#{key} " + value.join(' and ') }.join(', ')
     end
+
+    def validate_nested_inputs
+      validate_veteran
+      validate_service_information
+      validate_disabilities
+      validate_direct_deposit if directDeposit.present?
+    end
+
+    def validate_veteran
+      key = current_mailing_address
+      errors.add(:veteran, 'must include currentMailingAddress') unless veteran.key?(:currentMailingAddress)
+      errors.add(:currentMailingAddress, 'currentMailingAddress is not an object') unless key.is_a?(Hash)
+      validate_current_mailing_address unless errors[:currentMailingAddress].any?
+    end
```

Comment:
If `veteran.key?(:currentMailingAddress)` is false then `current_mailing_address` will be `nil`. Maybe we can use that property to eliminate the `key` alias altogether?

```
def validate_veteran
  errors.add(:veteran, 'must include currentMailingAddress') if current_mailing_address.nil?
  errors.add(:currentMailingAddress, 'currentMailingAddress is not an object') unless current_mailing_address.is_a?(Hash)
  validate_current_mailing_address unless errors[:currentMailingAddress].any?
end
```

Going a bit further, I would consider extracting these validation methods into a separate validation class. That would allow for accessor methods that dig into the nested structure, which would then lend themselves to rewriting these validations using the conventional approach. Very roughly:

```
class Form526AddressValidation
  extend Forwardable
  include ActiveModel::Validations

  def initialize(params)
    @params = params
  end

  def current_mailing_address
    @params.dig(:veteran, :currentMailingAddress)
  end

  delegate :addressLine1, :city, :state, :zipFirstFive, :country, :type, to: :current_mailing_address

  validates :current_mailing_address, presence: true, kind_of: Hash
  validates_presence_of :addressLine1, :city, :state, :zipFirstFive, :country, :type
end
```

That might be overkill. Hard to tell without trying it out. Just food for thought.

Reply:
Hey Drew, you are correct about the alias, honestly it was added because of Rubocop and line length.